pandas/core/common.py

```diff
@@ -1887,65 +1887,62 @@ def _maybe_box_datetimelike(value):
 
 _values_from_object = lib.values_from_object
 
-def _possibly_convert_objects(values, convert_dates=True,
-                              convert_numeric=True,
-                              convert_timedeltas=True):
+def _possibly_convert_objects(values,
+                              datetime=True,
+                              numeric=True,
+                              timedelta=True,
+                              coerce=False):
     """ if we have an object dtype, try to coerce dates and/or numbers """
 
-    # if we have passed in a list or scalar
+    conversion_count = sum((datetime, numeric, timedelta))
+    if conversion_count == 0:
+        import warnings
+        warnings.warn('Must explicitly pass type for conversion. Original '
```

Comment:
I think it should stay in until around 0.19.  For example, if you have a script that uses 

```
df.convert_objects()
```

in 0.16, you will get soft conversion to datetime.  If you run the same script in 0.17, you will get nothing, so it seems a warning is reasonable [Even if this code is not heavily used]